.golangci-kal.yml

```diff
@@ -0,0 +1,55 @@
+run:
+  timeout: 10m
+  go: "1.23"
+  allow-parallel-runners: true
+
+linters:
+  disable-all: true
+  enable:
+    - kal # linter for Kube API conventions
+   
+linters-settings:
+  custom:
+    kal:
+      type: "module"
+      description: KAL is the Kube-API-Linter and lints Kube like APIs based on API conventions and best practices.
+      settings:
+        linters:
+          enable:
+          - "conditions" # Ensure conditions have the correct json tags and markers.
+          - "commentstart" # Ensure comments start with the serialized version of the field name.
+          - "integers" # Ensure only int32 and int64 are used for integers.
+          - "jsontags" # Ensure every field has a json tag.
+          - "maxlength" # Ensure all strings and arrays have maximum lengths/maximum items.
+          - "nobools" # Bools do not evolve over time, should use enums instead.
+          - "nophase" # Phase fields are discouraged by the Kube API conventions, use conditions instead.
```

Comment:
This one will be controversial. We initially wanted to drop phase, but then later realized that the phases are actually useful to users because they quickly give users an idea about the phase a Cluster/Machine is in without having to parse through a number of conditions

Reply:
Interesting, we should check the history of the API convention on this one. As the comment says, the idea would generally be something like a Ready condition to show the general state, I guess if we can represent printer columns well based on the condition, then the phase may become less fundamental?

Do you happen to have a link to prior discussion on this topic?

Reply:
I'm not sure where that discussion happened. But I think @vincepri / @fabriziopandini should have context on it

Reply:
Given that recent discussion about diverging from kube conventions, and the consensus there, I think it makes sense to capture this, and disable the nophase linter and comment why we are disabling it